File: src/dotnet/Common/Models/ResourceProviders/Agent/AgentOrchestrationSettings.cs
using FoundationaLLM.Common.Constants.Agents;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.ResourceProviders.Agent
{
    /// <summary>
    /// Provides agent-related orchestration settings.
    /// </summary>
    public class AgentOrchestrationSettings
    {
        /// <summary>
        /// The agent's LLM orchestrator type.
        /// </summary>
        [JsonPropertyName("orchestrator")]
        public string? Orchestrator { get; set; }

        /// <summary>
        /// Dictionary with override values for the agent parameters.
        /// <para>
        /// For the list of supported keys, see <see cref="AgentParametersKeys"/>.
        /// </para>
        /// </summary>
        [JsonPropertyName("agent_parameters")]
        public Dictionary<string, object>? AgentParameters { get; set; }
    }
}

File: src/python/PythonSDK/foundationallm/models/agents/agent_orchestration_settings.py
from typing import Optional
from pydantic import BaseModel, Field

class AgentOrchestrationSettings(BaseModel):
    """
    Represents settings for the orchestration.
    """
    orchestrator: Optional[str] = Field(default='LangChain', description="The agent's LLM orchestrator type.")
    agent_parameters: Optional[dict] = Field(default=None, description="A dictionary containing override values for the agent parameters.")
